packages/foam-vscode/src/core/model/workspace.ts

```diff
@@ -112,9 +109,14 @@ export class FoamWorkspace implements IDisposable {
       resource = this.listByIdentifier(path)[0];
     } else {
       if (isAbsolute(path) || isSome(baseUri)) {
-        path = getExtension(path) ? path : path + '.md';
-        const uri = baseUri.resolve(path);
-        resource = uri ? this.resources.get(normalize(uri.path)) : null;
+        if (getExtension(path) !== '.md') {
```

Comment:
so basically we give priority to `.md` files, am I reading this right?

If there is a `note.pdf.md` and a `note.pdf`, `[[note.pdf]]` will resolve to the former.
If there is a `note2.pdf`, `[[note2.pdf]]` is resolved to the pdf file.

We might want to add a test case to document this

Reply:
There is a better description in https://github.com/foambubble/foam/issues/861#issuecomment-991951415

Reply:
I've added some new tests.